vendor/github.com/tudalex/seccomp-sandbox/sandbox.go

```diff
@@ -0,0 +1,98 @@
+struct sigaction old_action;
+void handler(int signum, siginfo_t *info, void *context) {
+	//TODO: Put this in a nice formatted json error
+    fprintf(stderr,"Jail violation caused by syscall %s. Code %d\n", syscall_mappings[info->si_syscall], info->si_syscall);
+    //fprintf(stderr,"Code %d\n", info->si_syscall);
+    exit(1);
+}
+
+void install_sighandler() {
+    struct sigaction action;
+    sigaction(SIGSYS, NULL, &action);
+    memset(&action, 0, sizeof action);
+    sigfillset(&action.sa_mask);
+    action.sa_handler = NULL;
+    action.sa_sigaction = handler;
+    action.sa_flags = SA_NOCLDSTOP | SA_SIGINFO | SA_ONSTACK;
+    sigaction(SIGSYS, &action, &old_action);
+}
+*/
+import "C"
+
+import (
+	"github.com/seccomp/libseccomp-golang"
+	"log"
+)
+
+type FilterAction string
+
+type FilterOperation struct {
+	FilterOn   []string
+	Action     seccomp.ScmpAction
+	Conditions []seccomp.ScmpCondition
+}
+
+type SandboxProfile struct {
+	DefaultPolicy seccomp.ScmpAction
+	Filters       []FilterOperation
+}
+
+func Jail(sandboxProfile SandboxProfile) {
+	C.install_sighandler()
+	filter, err := seccomp.NewFilter(sandboxProfile.DefaultPolicy)
+	if err != nil {
+		log.Fatal("Error creating filter: %s\n", err)
+	} else {
+		log.Printf("Created filter\n")
+	}
+	defer filter.Release()
+	action, err := filter.GetDefaultAction()
+	if err != nil {
+		log.Fatal("Error getting default action of filter\n")
+	} else if action != seccomp.ActTrap {
+		log.Printf("Default action of filter was set incorrectly!\n")
+	}
+	log.Printf("%s\n", sandboxProfile.DefaultPolicy)
+	for _, profileFilter := range sandboxProfile.Filters {
+		for _, callName := range profileFilter.FilterOn {
+			call, err := seccomp.GetSyscallFromName(callName)
+			if err != nil {
+				log.Fatal("Error getting syscall number of %s: %s\n", callName, err)
+			} else {
+				log.Printf("Got hook to syscall %d\n", call)
+			}
+			// if there are conditions, construct a conditional rule
+			if len(profileFilter.Conditions) > 0 {
+				err = filter.AddRuleConditional(call, profileFilter.Action, profileFilter.Conditions)
+				if err != nil {
+					log.Fatal("Error adding conditional rule: %s", err)
+				}
+			} else {
+				err = filter.AddRule(call, profileFilter.Action)
+				if err != nil {
+					log.Fatal("Error adding rule to restrict syscall: %s\n", err)
+				} else {
+					log.Printf("Added rule to restrict syscall %s\n", callName)
+				}
+			}
+		}
+	}
+	filter.SetTsync(true)
+	filter.SetNoNewPrivsBit(true)
+	err = filter.Load()
```

Comment:
In addition to this it could be useful to return some sort of status indication from this function to the caller so the agent could decide if it should continue if the jail failed, potentially based on a compile-time flag or configuration setting.